lms/models/h_group.py

```diff
@@ -1,61 +1,30 @@
-from typing import NamedTuple
+import sqlalchemy as sa
 
-from lms.models._hashed_id import hashed_id
+from lms.db import BASE
 
-MAX_GROUP_NAME_LENGTH = 25
 
+class HGroup(BASE):
+    UNIQUE_CONSTRAINT_NAME = "unique_h_group"
 
-class HGroup(NamedTuple):
-    name: str
-    authority_provided_id: str
-    type: str
+    __tablename__ = "h_group"
+    __table_args__ = (
+        sa.UniqueConstraint(
+            "name", "authority_provided_id", name=UNIQUE_CONSTRAINT_NAME
```

Comment:
I think you just omit the `name` here. There's a naming convention in `db/__init__.py` that gets applied automatically: https://github.com/hypothesis/lms/blob/d220299bd95b8f4d0262a45c169e35ab97948a1e/lms/db/__init__.py#L79